rust/src/io/commit/dynamodb.rs

```diff
@@ -0,0 +1,492 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! DynamoDB based external manifest store
+//!
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use aws_sdk_dynamodb::error::SdkError;
+use aws_sdk_dynamodb::operation::get_item::builders::GetItemFluentBuilder;
+use aws_sdk_dynamodb::operation::put_item::builders::PutItemFluentBuilder;
+use aws_sdk_dynamodb::operation::query::builders::QueryFluentBuilder;
+use aws_sdk_dynamodb::types::{AttributeValue, KeyType};
+use aws_sdk_dynamodb::Client;
+use snafu::OptionExt;
+
+use crate::error::{IOSnafu, NotFoundSnafu};
+use crate::io::commit::external_manifest::ExternalManifestStore;
+use crate::{Error, Result};
+
+impl<E> From<SdkError<E>> for Error {
+    fn from(e: SdkError<E>) -> Self {
+        Self::IO {
+            message: format!("dynamodb error: {e}"),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct DynamoDBExternalManifestStore {
```

Comment:
Could you write docstring here to describe the schema , primary / secondary key, or secondary index if needed.

Reply:
Also the consistency level, and potential conflicts. Basically, a brief design doc.